src/graph_populator.py

```
@@ -187,7 +189,7 @@ def construct_version_query(cls, input_json):
     @classmethod
     def construct_package_query(cls, input_json):
         """Construct the query to retrieve detailed information of given package."""
-        pkg_name = input_json.get('package')
+        pkg_name = input_json.get('package', '')
```

Technically it's OK, but, well, the question is 'what is better' ?
a) To somehow mask a missing piece to continue in any case (even if that means ingesting incorrect/incomplete data) ?
b) Or to fail early so that someone can check what's happened, fill and issue and fix the cause (even if that means more work) ?

:-)

Someone like me, who prefers (b) would actually do (the same as in my first comment)
`pkg_name = input_json['package']` so that you'd get an explicit `KeyError: 'package'` in case the input data happens to be incomplete.

---

👍

---

I made that change, because in the next line we are using the string contained in `pkg_name` to generate `tokens` in it.
Even better would be to first `get` the value and then check that it should not be `None` or `Empty` string, because either way it will be useless to ingest, and then throw an error.

---

Exactly.
You have couple options how to do the 'checking' part, I'd probably use simple [assert](https://docs.python.org/3/reference/simple_stmts.html#the-assert-statement):
```python
assert input_json.get('package'), "no or empty 'package'"
pkg_name = input_json['package']
```
This assert raises an `AssertionError` if the 'package' is not in `input_json` or the value there is `None` or `''`

You can even have it in a function as we do in [worker/utils](https://github.com/fabric8-analytics/fabric8-analytics-worker/blob/master/f8a_worker/utils.py#L184) or [worker/base](https://github.com/fabric8-analytics/fabric8-analytics-worker/blob/master/f8a_worker/base.py#L26)

---

is this PR still relevant?